docs/JestObjectAPI.md

```diff
@@ -119,6 +119,92 @@ test('implementation created by jest.genMockFromModule', () => {
 });
 ```
 
+This is how `genMockFromModule` will mock the follwing data types:
+
+#### `Function`
+
+A new function will be created. The new function will have no formal parameters and when called will return `undefined`. This functionality also applies to `async functions`.
```

Comment:
should mocked `async` functions return `Promise<undefined>` instead?

Reply:
I definitely think that makes more sense, I'll update it.

Reply:
yep, I meant that maybe we should change this. Good that the current behavior gets documented though!